# Lab 3 - EDA (Stats)
### Jackson Rolando

## Part 1: Review of Statistical Tests

My initial hypothesis:
The GPAs of students who play video games regularly are significantly lower than those who do not.

My answers to the questions:
1. A two-sample t-test is be used to compare a measured continuous variable between two groups. 
2. This situation is perfect for a two-sampled t-test. There are two groups, gamers and non-gamers, and we have the continuous variable of a GPA. 
Are there any particular assumptions that the t-test makes that may not hold here?
3. We don't know how our distributions fall. The test assumes a normal distribution, which we have no idea of confirming. The standard deviations are similar, but since they are close, and we have more than 30 samples in each category, the mismatch shouldn't be an issue.

**Null Hypothesesis:** There is not a significant difference between the groups.

**Alernative Hypothesesis:** There is a significant difference between the groups, and the gamers have a higher GPA. 

Here we apply the two-sampled t-test on the sample data: 
> 68 students said they play video games regularly, while 32 students said they did not. The 68 games have an average GPA of 3.4 with a standard deviation of 1.2, while the 32 non-gamers have an average GPA of 3.3 with a standard deviation of 1.1.

In [2]:
import scipy.stats as stats
stats.ttest_ind_from_stats(3.4, 1.2, 68, 3.3, 1.1, 32)

Ttest_indResult(statistic=0.39893881176878243, pvalue=0.6908062583072547)

The test yielded a p-value of 0.6908, way higher than the required maximum of 0.01. This means that we cannot reject the null hypothesis, as there is not a significant enough difference between the two groups of students.

The result of the t-test cannot prove my initial hypothesis either. My mentioning of one group having 'significantly lower' GPAs was just another way of saying there was a significant negative difference, which the t-test disproved. 

## Part 2: Exploring Additional Statistical Tests
a.  List  the  two  types  of  variables  for  which  the  test  is  appropriate.    Indicate  any 
assumptions that you would need to be aware of. 
 
b. Write down the general forms of the null and alternative hypotheses (one sentence per 
hypothesis). 
 
c. In your own words, write what it would mean if the test did and did not indicate 
statistical significance.

### Correlation and Linear Regression

A test for linear correlation is best done between two (usually) continuous features measured from the same sample. The assumptions usually made are less focused on the metrics of the data itself, but more on how the data was collected (random samples of a population vs specificly chosen independent variables) and/or controlled (potential confounding variables kept constant or intentionally made random). These assumptions/decisions affect what kinds of conclusions you can draw from the test, and which results you can use. 

A null hypothesis would predict the slope of the line of best fit between *insert variable 1* and *insert veriable 2* is zero, potentially with a threshold P-value.

An alternative hypothesis would hope that the line of best fit of a plot of *variable 1* and *variable 2* has a nonzero slope, potentially with a maximum P value of acceptance.

If the test indicates statistical significance, then depending on the decisions made when measuring samples (random samples from a population? chosen independent variables? controlled confounding variables?), you could use either the $r^2$ value or the P value for an implication of correlation or causation. For example, if you're measuring a correlation between two variables of random samples of a population with verifyably random confounding variables, then a very high $r^2$ value could imply causation. A relatively high correlation could mean that there is another (known or unknown) factor affecting both variables relatively predictably. There's something interesting going on. 

If the test does not indicate statistical significance between the variables of the correlation/regression, then one almost certainly does not cause the other, and there probably isn't another variable that affects the two variables similarly. 

### Kruskal-Wallis Test
This test is used to test for differences in a measured variable between different categories. Unlike other tests, it assumes <u>the different categories have a similar distribution</u>, but doesn't assume *which* distribution. It takes in the measurement variables not directly, but instead as a list of ranks, with the smallest measured value getting a rank of 1 and the largest value getting a rank of the number of total measured values. 

A null hypothesis predicts that there is no difference in the ranks of the means of each similarly-distributed group. An alternative hypothesis predicts that there is a significant difference between the means of the various categories.

If the test indicates significance, this indicates that the means of the measured variable are different between the groups, and the category may be a useful differentiator for the measured variable. If the test does not indicate significance, then the category probably isn't a great way to separate the sample for the measured variable. 

### Chi Squared Test

